src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -208,106 +224,169 @@ contract SecurityCouncilManager is Initializable, ArbitrumTimelock, ISecurityCou
+    /// @param _index index in securityCouncils of data to be remoed
+    function removeSecurityCouncil(uint256 _index) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        SecurityCouncilData storage securityCouncilToRemove = securityCouncils[_index];
+        SecurityCouncilData storage lastSecurityCouncil =
+            securityCouncils[securityCouncils.length - 1];
+
+        securityCouncils[_index] = lastSecurityCouncil;
+        securityCouncils.pop();
+        emit SecurityCouncilRemoved(
+            securityCouncilToRemove.securityCouncil,
+            securityCouncilToRemove.upgradeExecutor,
+            securityCouncilToRemove.updateAction,
+            securityCouncilToRemove.inbox
         );
-
-        this._scheduleDispatchUpdateMembersImpl(_membersData);
     }
 
-    function _scheduleDispatchUpdateMembersImpl(bytes calldata _membersData) public {
-        // TODO: less hacky way to do this?
-        require(msg.sender == address(this), "SecurityCouncilManager: not callable externally");
-        bytes32 salt = keccak256(abi.encodePacked(block.timestamp, block.number, _membersData));
-
-        TimelockControllerUpgradeable.schedule(
-            address(this), 0, _membersData, bytes32(0), salt, getMinDelay()
+    function _addSecurityCouncil(SecurityCouncilData memory _securityCouncilData) internal {
+        require(
+            _securityCouncilData.updateAction != address(0),
+            "SecurityCouncilManager: zero updateAction"
+        );
+        require(
+            _securityCouncilData.upgradeExecutor != address(0),
+            "SecurityCouncilManager: zero upgradeExecutor"
         );
-    }
 
-    /// @notice override timelock execute; forwards the message to the L1 router
-    function _execute(address __, uint256 ___, bytes calldata _membersData) internal override {
-        ArbSys(0x0000000000000000000000000000000000000064).sendTxToL1(
-            l1SecurityCouncilUpdateRouter, _membersData
+        require(
+            _securityCouncilData.securityCouncil != address(0),
+            "SecurityCouncilManager: zero securityCouncil"
+        );
+        securityCouncils.push(_securityCouncilData);
+        emit SecurityCouncilAdded(
+            _securityCouncilData.securityCouncil,
+            _securityCouncilData.upgradeExecutor,
+            _securityCouncilData.updateAction,
+            _securityCouncilData.inbox
         );
     }
+    /// @notice Set delay for messages to the L1 timelock. This should only be used to keep the minTimelockDelay value in sync with L1 (i.e., if the L1 side is updated, this should be too)
+    /// @param _minL1TimelockDelay new  L1 timelock delay value
 
-    /// @notice admin can update the l1SecurityCouncilUpdateRouter address
-    /// @param _l1SecurityCouncilUpdateRouter  new  l1SecurityCouncilUpdateRouter addresses
-    function setL1SecurityCouncilUpdateRouter(address _l1SecurityCouncilUpdateRouter)
+    function setMinL1TimelockDelay(uint256 _minL1TimelockDelay)
```

Comment:
I think lets just keep this as a DAO vote for now, but this would defo fall into the "permissionless message passing" framework thing we discussed once